base/string_utils.hpp

```diff
@@ -318,30 +318,30 @@ void ForEachMatched(string const & s, regex const & regex, TFn && fn)
 // form is chosen to fit both std::string and strings::UniString.
 // This function does not normalize either of the strings.
 template <typename TIter>
-uint32_t EditDistance(TIter const & b1, TIter const & e1, TIter const & b2, TIter const & e2)
+size_t EditDistance(TIter const & b1, TIter const & e1, TIter const & b2, TIter const & e2)
 {
   size_t n = distance(b1, e1);
   size_t m = distance(b2, e2);
-  vector<vector<uint32_t>> best(n + 1, vector<uint32_t>(m + 1, n + m));
-  best[0][0] = 0;
-  for (size_t i = 1; i <= n; ++i)
-    best[i][0] = i;
-  for (size_t j = 1; j <= m; ++j)
-    best[0][j] = j;
+  vector<size_t> prev(m + 1);
+  vector<size_t> cur(m + 1);
+  for (size_t j = 0; j <= m; j++)
+    prev[j] = j;
   auto it1 = b1;
   // 1-based to avoid corner cases.
   for (size_t i = 1; i <= n; ++i, ++it1)
   {
+    cur[0] = i;
     auto const & c1 = *it1;
     auto it2 = b2;
     for (size_t j = 1; j <= m; ++j, ++it2)
     {
       auto const & c2 = *it2;
 
-      best[i][j] = min(best[i - 1][j], best[i][j - 1]) + 1;
-      best[i][j] = min(best[i][j], best[i - 1][j - 1] + (c1 == c2 ? 0 : 1));
+      cur[j] = min(cur[j - 1], prev[j]) + 1;
+      cur[j] = min(cur[j], prev[j - 1] + (c1 == c2 ? 0 : 1));
     }
+    prev.swap(cur);
   }
-  return best[n][m];
+  return prev[m];
```

Comment:
No, notation [m] is better since it explicitly declares that we're interested in the prefix of the size m.